bench/sha256.bench.mo

```diff
@@ -0,0 +1,74 @@
+import Array "mo:base/Array";
+import Blob "mo:base/Blob";
+import Iter "mo:base/Iter";
+import Text "mo:base/Text";
+
+import Prim "mo:prim";
+
+import Bench "mo:bench";
+
+import Sha256 "../src/Sha256";
+
+module {
+  public func init() : Bench.Bench {
+    let bench = Bench.Bench();
+
+    bench.name("Sha256");
+    bench.description("Hash various message lengths from different types of input. Blocks are 64 bytes.");
+
+    let rows = [
+      "fromBlob",
+      "fromArray",
+      "fromIter",
+    ];
+    let cols = [
+      "0",
+      "1k blocks",
+      "1 MB",
+    ];
+
+    bench.rows(rows);
+    bench.cols(cols);
+
+    let rowSourceArrays : [[Nat8]] = [
+      [],
+      Array.tabulate<Nat8>(64_000, func(i) = 0x5f),
+      Array.tabulate<Nat8>(1_048_576, func(i) = 0x5f),
+    ];
+
+    let routines : [() -> ()] = Array.tabulate<() -> ()>(
+      rows.size() * cols.size(),
+      func(i) {
+        let row : Nat = i % rows.size();
+        let col : Nat = i / rows.size();
+
+        let source = rowSourceArrays[col];
+
+        switch (row) {
+          case (0) {
+            let blob = Blob.fromArray(source);
```

Comment:
This conversion should happen in advance, outside of the measured code.

But interesting to see that fromBlob is faster than fromArray even when this conversion is included. I did not expect that.

Reply:
It happens outside of measured code. See that next line returns function, which will be measured

Reply:
I see. Got it